# Mecánica de fluidos - Práctico 1

## Problema 1
Con referencia  a la figura y despreciando el rozamiento entre el pistón A y el cilindro que contiene el gas, determinar la presión manométrica en B en metros de columna de agua. Suponer el peso específico del gas 0,56 kg/m³.

![tp1-01.png](tp1-01.png)

### Solución
El área de $A = A_a = \pi r^2 = \pi \, (30 \, m)^2 $

In [3]:
import math
r = 60 / 2 # m
A_a = math.pi * r**2 # m^2
print('A_a =',round(A_a,2),'m^2')

A_a = 2827.43 m^2


La presión relativa bajo A es
$$ p_a = \frac{P_A}{A_a} $$

In [5]:
P_A = 1600 * 1000 # kgf peso del pistón
p_a = P_A / A_a # presión relativa bajo A
print('p_a =',round(p_a,2),'kgf/m^2')

p_a = 565.88 kgf/m^2


Para un gas ideal en condiciones isotérmicas
$$ p v = p_1 v_1 = R \qquad \qquad [1] $$
ahora como
$$ \gamma = g \delta \quad \Rightarrow \frac{1}{\delta} = \frac{g}{\delta} $$ y como 
$$ v = \frac{1}{\delta} \quad \Rightarrow \quad v = \frac{q}{\gamma} \qquad [2] $$
reemplazando [2] en [1] resulta:
$$ p \frac{g}{\gamma} = p_1 \frac{g_1}{\gamma_1} = R \qquad [3] $$
Suponiendo que el intervalo de elevación no es lo suficientemente grande como para que introduzca una variación significativa en $g$, de manera que $g$ puede considerarse como una constante. Luego, dividiendo por $g$ en [3]
$$ \frac{g}{\gamma} = \frac{p_1}{\gamma_1} = \frac{R}{g} = C \qquad [4] \qquad \therefore \quad \gamma = \frac{P}{C} \qquad [5] $$
Utilizando [5], puede expresarse la ecuación diferencial básica como sigue:
$$ \frac{dP}{dz} = \gamma = \frac{P}{C} $$
Separando variables e integrando entre $p_1$ y $p$ y entre $z_1$ y $z$, se tiene
## $$ \int_{p_1}^p \frac{dP}{P} = \int_{z_1}^z \frac{dz}{C} \qquad \Rightarrow \qquad \left. \ln p \right|_{p_1}^p = \left. \frac{z}{C} \right|_{z_1}^z \qquad \Rightarrow \qquad \ln \frac{p}{p_1} = \frac{1}{C} (z - z_1) $$
despejando $p$ y utilizando la ecuación [4] $p_1/ \gamma_1 = C$
## $$ \frac{p}{p_1} = e^{(\gamma_1 / p_1) (z-z_1)} $$
## $$ p = p_1 \, e^{(\gamma_1 / p_1) (z-z_1)} \qquad [7] $$
Como $p_1 = p_a = 565,88 \, kgf/m^2$ y $z_1 = 0$, $z = 70 \, m$ reemplazando en [7] resulta

In [10]:
p_1 = p_a
gamma = 0.56 # kgf/m^3
z = 70
z_1 = 0 # m
p_b = p_1 * math.e**(gamma/p_1*(z-z_1))
print('p_b =',round(p_b,2),'kgf/m^2')

p_b = 606.47 kgf/m^2


### $$ p_{H_2O} = \gamma_{H_2O} \cdot h = 1000 \, kgf/m^3 \cdot 1 \, m = 1000 \, kgf/m^2 = 1 \, mca $$
## $$ p_B = 606,47 \, kgf/m^2 = 0,606 \, mca = 60,64 \, cmca $$

In [1]:
%reset -f

## Problema 2
El aire del recipiente de la izquierda de la figura está a una presión de -23 cm de columna de mercurio. Determinar la cota del líquido manométrico en la parte derecha, en A.

![tp1-02.png](tp1-02.png)

### Solución
Considero que la presión dada para el aire en el recipiente de la izquierda es la presión a los 36m. Luego la presión en B es igual a la presión en A porque están al mismo nivel, (horizontal) y están interconectados por el mismo fluido estático.

    1 cmHg = 135,95 kgf/m²       1 kgf/cm² = 10000 kgf/m²

In [1]:
from sympy import *

In [4]:
z = symbols('z')
PB_aire = 135.95 * -23 # kgf/m^2
gamma_aceite = 0.8 * 1000 # kgf/m^3
h_aceite = 4 # m
gamma_l = 1.6 * 1000 # kgf/m^3
PA_aire = 2000 # kgf/m^3
gamma_agua = 1000 # kgf/m^3
h_agua = 1.5 # m
# Presión en A
PA = PA_aire  + gamma_agua*(h_agua + z) 
# Presión en B
PB = PB_aire + gamma_aceite*h_aceite + gamma_l*z  
# ecuación 1: PA = PB
eq1 = Eq(PA , PB)
# Resuelve ecuación 1, en z
seq1 = solve(eq1,z)
z = seq1[0]
print('z =',round(z,4),'m') 

z = 5.7114 m


In [8]:
h_A = 32 - z
print('Altura en A:')
print('h_A =',round(h_A,1),'m')

Altura en A:
h_A = 26.3 m


## Problema 3
Un tanque cilíndrico contiene agua hasta una altura de 50 mm. Dentro de éste se encuentra un tanque cilíndrico más pequeño, abierto, que contiene queroseno hasta una altura h, con una densidad de 0,8. Las siguientes presiones se conocen en los manómetros indicados: $p_B = 13,80 \, kPa$, $p_C = 13,82 \, kPa$. ¿Cuáles son la presión manométrica $p_A$ y la altura $h$ del queroseno? Suponga que se impide el movimiento del queroseno hacia la parte superior del tanque.
![tp1-03.png](tp1-03.png)

### Solución

$
\left\{
\begin{array}{l}
\mathrm{ 1 \, kgf/cm^2 = 98066,52 \, Pa = 98,07 \, kPa } \\
\mathrm{ 1 \, kgf/m^2 = 9,81 \, Pa = 9,81 \times 10^{-3} \, kPa}
\end{array}
\right.
$

### $$ p_A = p_C - \gamma_{H_2O} \cdot h_{H_2O} $$

In [4]:
g = 9.81 # m/s^2
p_C = (13.82*10**3) * (1/g) # kgf/m^2
gamma_agua = 1000 # kgf/m^3
h_agua = 50 / 1000 # m
p_A = p_C - gamma_agua * h_agua
print('p_A =',round(p_A,2),'kgf/m^2',' = ',round(p_A/98.07,2),'kPa')

p_A = 1358.77 kgf/m^2  =  13.86 kPa


__Cálculo de h__
### $
\begin{array}{l}
p_C - \gamma_{H_2O} \cdot h = p_B - \gamma_k \cdot h \\
p_C - p_B = (\gamma_{H_2O} - \gamma_k) \cdot h 
\end{array}
$
## $$ \Rightarrow \; h = \frac{p_C-p_B}{\gamma_{H_2O}-\gamma_k} $$

In [5]:
p_B = (13.8*10**3) * (1/g) # kgf/m^2
gamma_k = 0.8 * 1000 # kgf/m^3

h = (p_C - p_B)/(gamma_agua - gamma_k)
print('h =',round(h,5),'m')

h = 0.01019 m


## Ejercicio 4
¿Cuál es la presión manométrica dentro del tanque de aire?
![tp1-04.png](tp1-04.png)

Considero:

$
\begin{array}{l}
P_a = P_b \\
P_b = P_c \\
P_d = P_c - \gamma_{Hg} 200 \, mm \\
P_e = P_d + \gamma_{H_2O} 100 \, mm \\
P_f = P_e \\
P_f = \gamma_{Hg} 260 mm
\end{array}
$